rpc/namespaces/eth/api.go

```diff
@@ -571,14 +634,70 @@ func (api *PublicEthereumAPI) GetBlockByHash(hash common.Hash, fullTx bool) (map
 // GetBlockByNumber returns the block identified by number.
 func (api *PublicEthereumAPI) GetBlockByNumber(blockNum rpctypes.BlockNumber, fullTx bool) (map[string]interface{}, error) {
 	api.logger.Debug("eth_getBlockByNumber", "number", blockNum, "full", fullTx)
-	return api.backend.GetBlockByNumber(blockNum, fullTx)
+
+	if blockNum != rpctypes.PendingBlockNumber {
+		return api.backend.GetBlockByNumber(blockNum, fullTx)
+	}
+
+	height, err := api.backend.LatestBlockNumber()
+	if err != nil {
+		return nil, err
+	}
+
+	// latest block info
+	latestBlock, err := api.clientCtx.Client.Block(&height)
+	if err != nil {
+		return nil, err
+	}
+
+	unconfirmedTxs, err := api.clientCtx.Client.UnconfirmedTxs(100)
```

Comment:
what is the 100?

Reply:
number of pending txs queried from the mempool

Reply:
oof i was trying to copy the same value as `pendingTransactions` but missing a 0. Do you recommend something other than `1000` here?